designer/server/src/main/scala/pl/touk/nussknacker/ui/api/NodesResources.scala

```diff
@@ -114,21 +113,20 @@ class NodesResources(val processRepository: FetchingProcessRepository[Future],
               ParametersValidationResult(validationErrors = validationResults, validationPerformed = true)
             }
           }
-        } ~ path("suggestions") {
-          val modelData = typeToConfig.forTypeUnsafe(process.processingType)
-          val expressionSuggester = typeToExpressionSuggester.forTypeUnsafe(process.processingType)
-          implicit val typeDecoder: Decoder[TypingResult] = prepareTypingResultDecoder(modelData)
-          implicit val expressionSuggestionRequestDecoder: Decoder[ExpressionSuggestionRequest] = ExpressionSuggestionRequest.decoder(typeDecoder)
-
-          entity(as[ExpressionSuggestionRequest]) { expressionSuggestionRequest =>
-            complete {
-              expressionSuggester.expressionSuggestions(
-                expressionSuggestionRequest.expression,
-                expressionSuggestionRequest.caretPosition2d,
-                expressionSuggestionRequest.variables,
-              )
-            }
-          }
+        }
+      }
+    } ~ pathPrefix("parameters" / Segment / "suggestions") { processingType =>
+      val modelData = typeToConfig.forTypeUnsafe(processingType)
```

Comment:
What happens when we pass invalid `processingType`? We should return some sane error message

Reply:
Thanks, I've added handler for that but I've choose Not Found (since there is no processingType found) - WDYT?